pkgs/servers/ankisyncd/default.nix

```diff
@@ -1,67 +1,59 @@
-{ lib
-, fetchFromGitHub
-, python3
-, anki
-}:
+{ lib, stdenvNoCC, fetchFromGitHub, rustPlatform
+, protobuf }:
 
-python3.pkgs.buildPythonApplication rec {
+let
   pname = "ankisyncd";
-  version = "2.2.0";
-  src = fetchFromGitHub {
-    owner = "ankicommunity";
-    repo = "anki-sync-server";
-    rev = version;
-    sha256 = "196xhd6vzp1ncr3ahz0bv0gp1ap2s37j8v48dwmvaywzayakqdab";
+  version = "1.1.0";
+
+  # anki-sync-server-rs expects anki sources in the 'anki' folder
+  # of its own source tree, with a patch applied (mostly to make
+  # some modules public): bundle our own src derivation manually
+  # We apparently cannot do that in postPatch or equivalent in a
+  # buildRustPackage derivation because the cargo deps are processed
+  # first, there might be a better way of doing this? But it works
+  ankiSrc = fetchFromGitHub {
+    owner = "ankitects";
+    repo = "anki";
+    rev = "2.1.60";
+    hash = "sha256-hNrf6asxF7r7QK2XO150yiRjyHAYKN8OFCFYX0SAiwA=";
+    fetchSubmodules = true;
   };
-  format = "other";
 
-  installPhase = ''
-    runHook preInstall
-
-    mkdir -p $out/${python3.sitePackages}
-
-    cp -r ankisyncd utils ankisyncd.conf $out/${python3.sitePackages}
-    mkdir $out/share
-    cp ankisyncctl.py $out/share/
-
-    runHook postInstall
-  '';
-
-  fixupPhase = ''
-    PYTHONPATH="$PYTHONPATH:$out/${python3.sitePackages}:${anki}"
-
-    makeWrapper "${python3.interpreter}" "$out/bin/ankisyncd" \
-          --set PYTHONPATH $PYTHONPATH \
-          --add-flags "-m ankisyncd"
-
-    makeWrapper "${python3.interpreter}" "$out/bin/ankisyncctl" \
-          --set PYTHONPATH $PYTHONPATH \
-          --add-flags "$out/share/ankisyncctl.py"
-  '';
-
-  nativeCheckInputs = with python3.pkgs; [
-    pytest
-    webtest
-  ];
-
-  buildInputs = [ ];
-
-  propagatedBuildInputs = [ anki ];
+  src = stdenvNoCC.mkDerivation {
+    name = "anki-sync-server-rs-src";
+    src = fetchFromGitHub {
+      owner = "ankicommunity";
+      repo = "anki-sync-server-rs";
+      rev = version;
+      hash = "sha256-/sM2/ccIbOXJtjnBipfHgfnVTDNNI+zP/LT+CdSvQT8=";
+    };
+
+    dontConfigure = true;
+    dontBuild = true;
+    installPhase = ''
+      cp -r "$src/." "$out"
+      chmod +w "$out"
+      cp -r "${ankiSrc}" "$out/anki"
+      chmod -R +w "$out/anki"
+      patch -d "$out/anki" -Np1 < "$src/anki_patch/d9d36078f17a2b4b8b44fcb802eb274911ebabe7_anki_rslib.patch"
+    '';
+  };
+in rustPlatform.buildRustPackage rec {
```

Comment:
The `rec` there isn't needed (thanks [nil](https://github.com/oxalica/nil))

Reply:
Thanks! I need to setup some LSP.